R/annotation-ticks.R

```diff
@@ -0,0 +1,351 @@
+              delog = delog[s])
+
+            if (scaled) {
+                if (!delog[s])
+                  yticks$value <- log(yticks$value, base[s])
+            }
+
+            names(yticks)[names(yticks) == "value"] <- "y"  # Rename to 'y' for coordinates$transform
+            yticks <- coord$transform(yticks, panel_scales)
+
+            # Make the grobs
+            if (grepl("l", sides[s])) {
+                ticks$y_l <- with(data, grid::segmentsGrob(y0 = grid::unit(yticks$y,
+                  "native"), y1 = grid::unit(yticks$y, "native"), x0 = grid::unit(yticks$start,
+                  "cm"), x1 = grid::unit(yticks$end, "cm"), gp = grid::gpar(col = alpha(colour,
+                  alpha), lty = linetype, lwd = size * .pt)))
+            }
+            if (grepl("r", sides[s])) {
+                ticks$y_r <- with(data, grid::segmentsGrob(y0 = grid::unit(yticks$y,
+                  "native"), y1 = grid::unit(yticks$y, "native"), x0 = grid::unit(1,
+                  "npc") - grid::unit(yticks$start, "cm"), x1 = grid::unit(1, "npc") -
+                  grid::unit(yticks$end, "cm"), gp = grid::gpar(col = alpha(colour,
+                  alpha), lty = linetype, lwd = size * .pt)))
+            }
+        }
+    }
+    grid::gTree(children = do.call("gList", ticks))
+}, default_aes = aes(colour = "black", size = 0.5, linetype = 1, alpha = 1))
+
+
+# Calculate the position of log tick marks Returns data frame with: - value: the
+# position of the log tick on the data axis, for example 1, 2, ..., 9, 10, 20, ...
+# - start: on the other axis, start position of the line (usually 0) - end: on the
+# other axis, end position of the line (for example, .1, .2, or .3)
+calc_ticks <- function(
+  base = 10,
+  ticks_per_base = base - 1,
+  minpow = 0,
+  maxpow = minpow + 1,
+  majorTicks = 0,
+  start = 0,
+  shortend = 0.1,
+  midend = 0.2,
+  longend = 0.3,
+  delog = FALSE){
+
+    # Number of blocks of tick marks
+    reps <- maxpow - minpow
+
+    # For base 10: 1, 2, 3, ..., 7, 8, 9, 1, 2, ...
+    ticknums <- rep(seq(1, base - 1, length.out = ticks_per_base), reps)
+
+    # For base 10: 1, 1, 1, ..., 1, 1, 1, 2, 2, ... (for example)
+    powers <- rep(seq(minpow, maxpow - 1), each = ticks_per_base)
+
+    ticks <- ticknums * base^powers
+
+    ticks <- c(ticks, base^maxpow)  # Add the last tick mark
+
+    # Set all of the ticks short
+    tickend <- rep(shortend, length(ticks))
+
+    # Get the position within each cycle, 0, 1, 2, ..., 8, 0, 1, 2. ...
+    cycleIdx <- ticknums - 1
+
+    # Set the 'major' ticks long
+    tickend[cycleIdx == 0] <- longend
+
+    # Where to place the longer tick marks that are between each base For base 10, this
+    # will be at each 5
+    longtick_after_base <- floor(ticks_per_base/2)
+    tickend[cycleIdx == longtick_after_base] <- midend
+
+    if (delog) {
+
+        ticksCopy = ticks
+
+        regScale = log(ticks, base)
+
+        majorTicks = sort(unique(c(minpow,
```

Comment:
I believe these function calls should be on separate lines, though I'm not 100% sure